### PR TITLE
Document all the `child_process` options - fixes #90

### LINKED ISSUE #90: Document all the `child_process` options
People don't seem to notice `execa` supports all the `child_process` options. Would be nice to have the documentation for them here.

### PR DESCRIPTION
**Pull request by SamVerschueren**
I documented all the `child_process` options as described in #90. Feedback appreciated :)!

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -108,7 +108,59 @@ Returns the same result object as [`child_process.spawnSync`](https://nodejs.org
 
 ### options
 
-Additional options:
+Type: `object`
```
Comment (sindresorhus) [inline readme.md]: `object` => `Object`

**Comment by sindresorhus** [comment]
With this, you can remove all the `Same options as ...` sentences.

You also need to document the following:

```
encoding <string> (Default: 'utf8')
timeout <number> (Default: 0)
maxBuffer <number> Largest amount of data in bytes allowed on stdout or stderr.
```

**Comment by SamVerschueren** [comment]
What about `killSignal`?

```
killSignal <string> | <integer> The signal value to be used when the spawned process will be killed. (Default: 'SIGTERM')
```

**Comment by SamVerschueren** [comment]
What about `killSignal`?

```
killSignal <string> | <integer> The signal value to be used when the spawned process will be killed. (Default: 'SIGTERM')
```

**Comment by kevva** [comment]
> What about killSignal?

Good catch. Actually, I don't think that one works as it stands https://github.com/sindresorhus/execa/blob/master/index.js#L160. `parsed.killSignal` should be `parsed.opts.killSignal` I think.

**Comment by SamVerschueren** [comment]
I think you are right. The `parsed` object doesn't return `killSignal` like that... https://github.com/sindresorhus/execa/blob/master/index.js#L47

**Comment by sindresorhus** [comment]
🙌

**Pull request merged**